ladybug/location.py

```diff
@@ -140,37 +142,40 @@ def from_idf(cls, idf_string):
 
     @property
     def latitude(self):
-        """Location latitude."""
+        """Get or set the location latitude in degrees."""
         return self._lat
 
     @latitude.setter
     def latitude(self, lat):
         self._lat = 0 if not lat else float(lat)
-        assert -90 <= self._lat <= 90, "latitude should be between -90..90."
+        assert -90 <= self._lat <= 90, \
+            'latitude must be between -90 and 90. Got {}.'.format(self._lat)
 
     @property
     def longitude(self):
-        """Location longitude."""
+        """Get or set the location longitude in degrees."""
         return self._lon
 
     @longitude.setter
     def longitude(self, lon):
         self._lon = 0 if not lon else float(lon)
-        assert -180 <= self._lon <= 180, "longitude should be between -180..180."
+        assert -180 <= self._lon <= 180, \
+            'longitude must be between -180 and 180. Got {}.'.format(self._lon)
 
     @property
     def time_zone(self):
-        """Location time zone."""
+        """Get or set the location time zone as a number between -12 and + 12."""
         return self._tz
 
     @time_zone.setter
     def time_zone(self, tz):
-        self._tz = 0 if not tz else float(tz)
-        assert -12 <= self._tz <= 12, "Time zone should be between -12.0..12.0"
+        self._tz = round(self._lon / 15) if tz is None else float(tz)
+        assert -12 <= self._tz <= 12, \
```

Comment:
Oh, wow. You're right!  Because of the shape of the international dateline, there's a significant part of the Pacific ocean that goes past +12:
https://www.timeanddate.com/time/map/
It looks like mainland NZ is still in +12 but some of the islands off the coast that dip into this region. Ok, I will correct this now for both the Location and the Sunpath.

Reply:
Done!